### PR TITLE
[examples] Add bespoke JS version of Context Basics

### PR DESCRIPTION
**Pull request by augustjk**
Fixes https://github.com/lit/lit.dev/issues/1232

Preview: https://pr1233-15216bd---lit-dev-5ftespv5na-uc.a.run.app/playground/#sample=examples/context-basics
(toggle to JS)

### CODE REVIEW

File: packages/lit-dev-content/samples/examples/context-basics/elements.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   */
+  set data(value) {
+    this._data = value;
+    this._provider.setValue(value);
+    this.requestUpdate();
+  }
+
+  get data() {
+    return this._data;
+  }
```
Comment (augustjk) [inline packages/lit-dev-content/samples/examples/context-basics/elements.js:38]: I considered doing
```js
  willUpdate(changedProperties) {
    if (changedProperties.has('data')) {
      this._provider.setValue(this.data);
    }
  }
```
but there was a subtle timing issue that `willUpdate()` will get called after the consumer is registered so the initial value the consumer gets could never be the data set as attribute on the `<provider-el>`. The subscriber would have to consume with `{subscribe: true}` to get the same behavior as the TS version.

Comment (AndrewJakubowicz) [reply packages/lit-dev-content/samples/examples/context-basics/elements.js:38]: That's interesting!

File: packages/lit-dev-content/samples/examples/context-basics/elements.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    this.data = 'Initial';
+  }
+
+  /**
+   * `data` will be provided to any consumer that is in the DOM tree below it.
+   */
+  set data(value) {
+    this._data = value;
+    this._provider.setValue(value);
+    this.requestUpdate();
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/samples/examples/context-basics/elements.js:33]: With Lit 3, I think `requestUpdate` will always be called.

Comment (augustjk) [reply packages/lit-dev-content/samples/examples/context-basics/elements.js]: Nice catch! Fixed!

**Pull request merged**